hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -248,12 +261,32 @@ public static Writable avroToArrayWritable(Object value, Schema schema) {
     }
   }
 
+  public static Schema readSchemaFromLogFile(FileSystem fs, Path path) throws IOException {
+    Reader reader = HoodieLogFormat.newReader(fs, new HoodieLogFile(path), null);
+    HoodieAvroDataBlock lastBlock = null;
+    while (reader.hasNext()) {
+      HoodieLogBlock block = reader.next();
+      if (block instanceof HoodieAvroDataBlock) {
+        lastBlock = (HoodieAvroDataBlock) block;
+      }
+    }
+    if (lastBlock != null) {
+      return lastBlock.getSchema();
+    }
+    return null;
+  }
+
   /**
    * Goes through the log files and populates a map with latest version of each key logged, since
    * the base split was written.
    */
   private void init() throws IOException {
     writerSchema = new AvroSchemaConverter().convert(baseFileSchema);
+    if (split.getDeltaFilePaths().size() > 0) {
```

Comment:
this is temporary. I plan to reimplement this nicely. 

Which change in avro/parquet deps, record reader test started to fail since the "name" property was not properly retained in the base parquet' file's schema.  This caused avro deser to fail on the log data. But, this was anyway a stop gap since we should be reading schema from the log file and not base if we want latest columns to show up. So, just did that and it in turn ended up reading schema from teh data block itself, which fixed the original test failure . basically, fixed another bug and the test issue went away

Reply:
Sounds good